Core/Utilities/MySQL.py

```diff
@@ -463,17 +478,29 @@ def __escapeString( self, myString ):
       return retDict
     connection = retDict['Value']
 
-    specialValues = ( 'UTC_TIMESTAMP', 'TIMESTAMPADD', 'TIMESTAMPDIFF' )
-
     try:
       myString = str( myString )
     except ValueError:
       return S_ERROR( "Cannot escape value!" )
 
+    timeUnits = [ 'MICROSECOND', 'SECOND', 'MINUTE', 'HOUR', 'DAY', 'WEEK', 'MONTH', 'QUARTER', 'YEAR' ]
+
     try:
-      for sV in specialValues:
-        if myString.find( sV ) == 0:
-          return S_OK( myString )
+      # Check datetime functions first
+      if myString.strip() == 'UTC_TIMESTAMP()':
+        return S_OK( myString )
+
+      for func in [ 'TIMESTAMPDIFF', 'TIMESTAMPADD' ]:
+        if myString.strip().startswith( '%s(' % func ) and myString.strip().endswith( ')' ):
+          args = myString.strip()[:-1].replace( '%s(' % func, '' ).strip().split(',')
```

Comment:
That's why the string is stripped before, it must end with ')' if it is a legal function call, and this is checked in the previous if statement